Mathlib/Combinatorics/SimpleGraph/Matching.lean

```diff
@@ -309,4 +317,75 @@ lemma exists_maximal_isMatchingFree [Finite V] (h : G.IsMatchingFree) :
   obtain ⟨Gmax, hGmax⟩ := Finite.exists_le_maximal h
   exact ⟨Gmax, ⟨hGmax.1, ⟨hGmax.2.prop, fun _ h' ↦ hGmax.2.not_prop_of_gt h'⟩⟩⟩
 
+/-- A graph `G` consists of a set of cycles, if each vertex is either isolated or connected to
+exactly two vertices. This is used to create new matchings by taking the `symmDiff` with cycles.
+The definition of `symmDiff` that makes sense is the one for `SimpleGraph`. This is why this
+definition is for graphs, rather than subgraphs.
+-/
+def IsCycles (G : SimpleGraph V) := (∀ v : V, (G.neighborSet v) = ∅ ∨ (G.neighborSet v).ncard = 2)
+
+/--
+Given a vertex with one edge in a graph of cycles this gives the other edge incident
+to the same vertex.
+-/
+lemma IsCycles.other_adj_of_adj (h : G.IsCycles) (hadj : G.Adj v w) :
+    ∃ w', w ≠ w' ∧ G.Adj v w' := by
+  simp_rw [← SimpleGraph.mem_neighborSet] at hadj ⊢
+  cases' h v with hl hr
+  · exact ((Set.eq_empty_iff_forall_not_mem.mp hl) _ hadj).elim
+  · obtain ⟨w', hww'⟩ := (G.neighborSet v).exists_ne_of_one_lt_ncard (by omega) w
+    exact ⟨w', ⟨hww'.2.symm, hww'.1⟩⟩
+
+lemma Subgraph.IsPerfectMatching.symmDiff_spanningCoe_IsCycles
+    {M : Subgraph G} {M' : Subgraph G'} (hM : M.IsPerfectMatching)
+    (hM' : M'.IsPerfectMatching) : (symmDiff M.spanningCoe M'.spanningCoe).IsCycles := by
+  intro v
+  obtain ⟨w, hw⟩ := hM.1 (hM.2 v)
+  obtain ⟨w', hw'⟩ := hM'.1 (hM'.2 v)
+  simp only [symmDiff_def, Set.eq_empty_iff_forall_not_mem, SimpleGraph.mem_neighborSet,
+    SimpleGraph.sup_adj, sdiff_adj, spanningCoe_adj, not_or, not_and, not_not, Set.ncard_eq_two,
+    ne_eq]
+  by_cases hww' : w = w'
+  · simp_all [hww']
+  · right
+    use w, w'
+    aesop
+
+/--
+A graph `G` is alternating with respect to some other graph `G'`, if exactly every other edge in
```

Comment:
```suggestion
A graph `G` made up of cycles is alternating with respect to some other graph `G'`, if exactly every other edge in
```
because "every other edge" doesn't even make sense if this isn't the case, right?

Maybe `G.IsAlternating G'` should even assume `G.IsCycles`?

Reply:
This is why I put "degree bounded by 2" (changed to `at most` because I think that's clearer). This could also be applied for augmenting paths, which can be used to extend matchings. (And in which case the endpoints of the paths have degree 1).